### ISSUE TITLE
Allow to cancel goals while doing motion planning

### ISSUE THREAD

**Issue by adolfo-rt** [pal-robotics/play_motion#31]
This is what happens when an action goal is interrupted while planning:

```
To transition to a succeeded state, the goal must be in a preempting or active state, it is currently in state: 8
```

State 8 is RECALLED

**Comment by po1** [comment]
Are you setting the goal to `setAccepted` before launching the planning?

**Comment by adolfo-rt** [comment]
I pushed a partial fix in fa6dcb30984a09dfefaf96d6fc624ca7b068837e. Currently `setAccepted` is still called after planning. I need to think about how to reorganize things to accept goals earlier. This will likely walk down the path of adding threads and synchronization :-(.

**Comment by adolfo-rt** [comment]
The above fix should be enough to get us moving. The problem of not being able to preempt motion planning needs to be addressed upstream, and is being tracked here ros-planning/moveit_ros#439.
